File: src/main/java/store/wckd/server/controller/MeController.java
package store.wckd.server.controller;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import store.wckd.server.entity.User;
import store.wckd.server.service.UserService;

@RestController
public class MeController {

    @GetMapping("/me")
    public User me() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();

        return (User) authentication.getPrincipal();
    }

}

File: src/main/java/store/wckd/server/controller/MeController.kt
package store.wckd.server.controller

import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import store.wckd.server.dto.UserResponseDTO
import store.wckd.server.entity.User

@RestController
class MeController {
    @GetMapping("/me")
    fun me(): UserResponseDTO {
        val authentication = SecurityContextHolder.getContext().authentication

        return (authentication.principal as User).toDTO()
    }
}